lib/UBOTypeTransformPass.cpp

```diff
@@ -332,7 +329,7 @@ bool clspv::UBOTypeTransformPass::RemapUser(User *user, Module &M) {
   bool changed = RemapValue(user, M);
 
   for (Use &use : user->operands()) {
-    User *operand_user = use.getUser();
+    User *operand_user = dyn_cast<User>(use.get());
```

Comment:
Previously this just returned User the function was called with! Didn't cause problems because it was generally never acted on due to the filter below. Now it properly traverses constant operands. This gets covered by many existing tests using the new data type in the resource function. Wasn't a problem until the SPIRVProducerPass starting using those types.